src/needless_bool.rs

```diff
@@ -78,6 +93,57 @@ impl LateLintPass for NeedlessBool {
     }
 }
 
+#[derive(Copy,Clone)]
+pub struct BoolComparison;
+
+impl LintPass for BoolComparison {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(BOOL_COMPARISON)
+    }
+}
+
+impl LateLintPass for BoolComparison {
+    fn check_expr(&mut self, cx: &LateContext, e: &Expr) {
+        if let ExprBinary(Spanned{ node: BiEq, .. }, ref left_side, ref right_side) = e.node {
+            match (fetch_bool_expr(left_side), fetch_bool_expr(right_side)) {
+                (Some(true), None) => {
+                    let side_snip = snippet(cx, right_side.span, "..");
+                    let hint = format!("`{}`", side_snip);
+                    span_lint(cx,
+                              BOOL_COMPARISON,
+                              e.span,
+                              &format!("you can simplify this boolean comparison to {}", hint));
```

Comment:
In that case, it may be good to revise how the suggestions on the `needless_bool` lint are displayed as well. That is what I used for reference originally, e.g.:

    span_lint(cx,
        NEEDLESS_BOOL,
        e.span,
        &format!("you can reduce this if-then-else expression to just {}", hint));